registry/auth/htpasswd/access.go

```diff
@@ -32,18 +36,7 @@ func newAccessController(options map[string]interface{}) (auth.AccessController,
 		return nil, fmt.Errorf(`"path" must be set for htpasswd access controller`)
 	}
 
-	f, err := os.Open(path.(string))
-	if err != nil {
-		return nil, err
-	}
-	defer f.Close()
-
-	h, err := newHTPasswd(f)
-	if err != nil {
-		return nil, err
-	}
-
-	return &accessController{realm: realm.(string), htpasswd: h}, nil
+	return &accessController{realm: realm.(string), path: path.(string), modtime: "", mu: sync.Mutex{}, htpasswd: nil}, nil
```

Comment:
Nit: `mu: sync.Mutex{}` and `htpasswd: nil` aren't necessary. These fields will default to the zero value.